world.go

```diff
@@ -32,6 +32,13 @@ type World interface {
 	StorageAccessor() StorageAccessor
 	// ArcheTypes returns the archetypes in the world.
 	Archetypes() []*storage.Archetype
+
+	// OnCreate registers a callback function that gets triggered when an entity is created.
+	OnCreate(callback func(entity Entity))
```

Comment:
@im-gin Can we add the World as a parameter to the callback function? We might need to access the world in case we want to get the Entry object when we extend this feature to implement a more generic event system, as you mentioned in the PR comment.

Reply:
Oh yeah, very good point!